account_loan/models/account_loan_line.py

```diff
@@ -0,0 +1,447 @@
+            "date": self.date,
+            "ref": self.name,
+            "journal_id": self.loan_id.journal_id.id,
+            "line_ids": [(0, 0, vals) for vals in self._get_long_term_move_line_vals()],
+        }
+
+    def _generate_invoice(self):
+        """
+        Computes invoices of leases
+        :return: list of account.move generated
+        """
+        res = []
+        for record in self:
+            if not record.move_ids:
+                if record.loan_id.line_ids.filtered(
+                    lambda r: r.date < record.date and not r.move_ids
+                ):
+                    raise UserError(_("Some invoices must be created first"))
+                invoice = self.env["account.move"].create(record._invoice_vals())
+                res.append(invoice.id)
+                for line in invoice.invoice_line_ids:
+                    line.tax_ids = line._get_computed_taxes()
+                invoice.flush_recordset()
+                if record.loan_id.post_invoice:
+                    invoice.action_post()
+                if (
+                    record.long_term_loan_account_id
+                    and record.long_term_principal_amount != 0
+                ):
+                    move = self.env["account.move"].create(
+                        record._long_term_move_vals()
+                    )
+                    if record.loan_id.post_invoice:
+                        move.action_post()
+                    res.append(move.id)
+        return res
+
+    def _get_long_term_move_line_vals(self):
+        return [
+            {
+                "account_id": self.loan_id.short_term_loan_account_id.id,
+                "credit": self.long_term_principal_amount,
+                "debit": 0,
+            },
+            {
+                "account_id": self.long_term_loan_account_id.id,
+                "credit": 0,
+                "debit": self.long_term_principal_amount,
+            },
+        ]
+
+    def view_account_values(self):
+        """Shows the invoice if it is a leasing or the move if it is a loan"""
+        self.ensure_one()
+        if self.is_leasing:
+            return self.view_account_invoices()
+        return self.view_account_moves()
+
+    def view_process_values(self):
+        """Computes the annuity and returns the result"""
+        self.ensure_one()
+        if self.is_leasing:
+            self._generate_invoice()
+        else:
+            self._generate_move()
+        return self.view_account_values()
+
+    def view_account_moves(self):
+        self.ensure_one()
+        result = self.env["ir.actions.act_window"]._for_xml_id(
+            "account.action_move_line_form"
+        )
+        result["context"] = {
+            "default_loan_line_id": self.id,
+            "default_loan_id": self.loan_id.id,
+        }
+        result["domain"] = [("loan_line_id", "=", self.id)]
+        if len(self.move_ids) == 1:
+            res = self.env.ref("account.move.form", False)
```

Comment:
I think the reference should be
```suggestion
            res = self.env.ref("account.view_move_form", False)
```

Reply:
Nice catch